src/library_gl.js

```diff
@@ -2340,6 +2345,87 @@ var LibraryGL = {
     GLctx.uniform4fv(location, view);
   },
 
+#if USE_WEBGL2
+  glUniform1ui__sig: 'vii',
+  glUniform1ui: function(location, v0) {
+#if GL_ASSERTIONS
+    GL.validateGLObjectID(GL.uniforms, location, 'glUniform1ui', 'location');
+#endif
+    location = GL.uniforms[location];
+    GLctx.uniform1ui(location, v0);
+  },
+
+  glUniform2ui__sig: 'viii',
+  glUniform2ui: function(location, v0, v1) {
+#if GL_ASSERTIONS
+    GL.validateGLObjectID(GL.uniforms, location, 'glUniform2ui', 'location');
+#endif
+    location = GL.uniforms[location];
+    GLctx.uniform2ui(location, v0, v1);
+  },
+
+  glUniform3ui__sig: 'viiii',
+  glUniform3ui: function(location, v0, v1, v2) {
+#if GL_ASSERTIONS
+    GL.validateGLObjectID(GL.uniforms, location, 'glUniform3ui', 'location');
+#endif
+    location = GL.uniforms[location];
+    GLctx.uniform3ui(location, v0, v1, v2);
+  },
+
+  glUniform4ui__sig: 'viiiii',
+  glUniform4ui: function(location, v0, v1, v2, v3) {
+#if GL_ASSERTIONS
+    GL.validateGLObjectID(GL.uniforms, location, 'glUniform4ui', 'location');
+#endif
+    location = GL.uniforms[location];
+    GLctx.uniform4ui(location, v0, v1, v2, v3);
+  },
+
+  glUniform1uiv__sig: 'viii',
+  glUniform1uiv: function(location, count, value) {
+#if GL_ASSERTIONS
+    GL.validateGLObjectID(GL.uniforms, location, 'glUniform1uiv', 'location');
+#endif
+    location = GL.uniforms[location];
+    value = {{{ makeHEAPView('32', 'value', 'value+count*4') }}};
```

Comment:
we could probably just write `HEAPU32.subarray` etc. here, could be clearer. `makeHEAPView` is not that useful anymore (it used to be, when we had other HEAP models). But, I'm fine either way, whichever you feel is nicer.